general-superstaq/general_superstaq/check/check_utils.py

```diff
@@ -298,7 +294,7 @@ def get_file_parser() -> argparse.ArgumentParser:
     )
 
     help_text = "The files to check. If not passed any files, inspects the entire repo."
-    parser.add_argument("files", nargs="*", help=help_text)
+    parser.add_argument("--files", nargs="*", default=[], help=help_text)
```

Comment:
This appeared to be necessary to get rid of warnings like this:
![image](https://github.com/Infleqtion/client-superstaq/assets/567092/2d0a5531-99bc-4184-86b6-17f6ef1cbf02)

Reply:
this changes our usage pretty drastically - to pass files we'll have to use e.g. `./check/pytest_.py --files x.py ...` instead of  `./check/pytest_.py x.py ...`

it seems like this warning only comes up when using `parse_known_intermixed_args` - can we just use `parse_known_args` everywhere instead? (and also add `action="extend"` in each case, to prevent sub-parsers from clobbering `parsed_args.files`)

Reply:
Oh right duh.  Sorry that was dumb :sweat_smile:

Reply:
I like adding `--skip` to the common parser.